rasa/nlu/emulators/luis.py

```diff
@@ -1,50 +1,77 @@
 from typing import Any, Dict, Text
 
 from rasa.nlu.emulators.emulator import Emulator
+from rasa.shared.nlu.constants import (
+    ENTITIES,
+    ENTITY_ATTRIBUTE_TYPE,
+    ENTITY_ATTRIBUTE_ROLE,
+    ENTITY_ATTRIBUTE_VALUE,
+    ENTITY_ATTRIBUTE_END,
+    ENTITY_ATTRIBUTE_START,
+    INTENT_RANKING_KEY,
+    TEXT,
+    INTENT,
+    INTENT_NAME_KEY,
+    PREDICTED_CONFIDENCE_KEY,
+)
 from typing import List, Optional
 
 
 class LUISEmulator(Emulator):
-    """Emulates Luis responses."""
+    """Emulates the response format of the LUIS Endpoint API v3.0 /predict endpoint.
+
+    https://westcentralus.dev.cognitive.microsoft.com/docs/services/luis-endpoint-api-v3-0/
+    """
 
     def _top_intent(self, data) -> Optional[Dict[Text, Any]]:
-        if data.get("intent"):
-            return {
-                "intent": data["intent"]["name"],
-                "score": data["intent"]["confidence"],
-            }
-        else:
+        intent = data.get(INTENT)
+
+        if not intent:
             return None
 
-    def _ranking(self, data) -> List[Dict[Text, Any]]:
-        if data.get("intent_ranking"):
+        return {
+            "intent": intent[INTENT_NAME_KEY],
+            "score": intent[PREDICTED_CONFIDENCE_KEY],
+        }
+
+    def _intents(self, data) -> List[Dict[Text, Any]]:
+        if data.get(INTENT_RANKING_KEY):
             return [
-                {"intent": el["name"], "score": el["confidence"]}
-                for el in data["intent_ranking"]
+                {"intent": el[INTENT_NAME_KEY], "score": el[PREDICTED_CONFIDENCE_KEY]}
+                for el in data[INTENT_RANKING_KEY]
             ]
         else:
             top = self._top_intent(data)
             return [top] if top else []
 
     def normalise_response_json(self, data: Dict[Text, Any]) -> Dict[Text, Any]:
-        """Transform data to luis.ai format."""
+        """Transform response JSON to LUIS format.
 
-        top_intent = self._top_intent(data)
-        ranking = self._ranking(data)
+        Args:
+            data: input JSON data as a dictionary.
+
+        Returns:
+            The transformed input data.
+        """
         return {
-            "query": data["text"],
-            "topScoringIntent": top_intent,
-            "intents": ranking,
-            "entities": [
-                {
-                    "entity": e["value"],
-                    "type": e["entity"],
-                    "startIndex": e.get("start"),
-                    "endIndex": (e["end"] - 1) if "end" in e else None,
-                    "score": e.get("confidence"),
-                }
-                for e in data["entities"]
-            ]
-            if "entities" in data
-            else [],
+            "query": data[TEXT],
+            "prediction": {
+                "topIntent": self._top_intent(data),
+                "intents": self._intents(data),
+                "entities": [
```

Comment:
according to the linked API documentation this returns a `Dict`  of `List[List[Text]]`. (no idea why the list is nested 😆 )